packages/react-charting/src/utilities/utilities.ts

```diff
@@ -549,6 +696,83 @@ export function createWrapOfXLabels(wrapLabelProps: IWrapLabelProps) {
   return removeVal > 0 ? removeVal : 0;
 }
 
+/**
+ * This method used for wrapping of y axis labels (tick values).
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function createYAxisLabels(
+  node: SVGElement | null,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  yAxis: any,
+  noOfCharsToTruncate: number,
+  truncateLabel: boolean,
+  xValue: number,
+  isRtl: boolean,
+) {
+  if (node === null) {
+    return;
+  }
+  const axisNode = d3Select(node).call(yAxis);
+  axisNode.selectAll('.tick text').each(function () {
+    const text = d3Select(this);
+    const totalWord = text.text();
+    const truncatedWord = `${text.text().slice(0, noOfCharsToTruncate)}...`;
+    const totalWordLength = text.text().length;
+    const padding = truncateLabel ? 1.5 : 1; // ems
+    const y = text.attr('y');
+    const x = text.attr('x');
+    const dy = parseFloat(text.attr('dy'));
+    const dx = 0;
+    text
+      .text(null)
+      .append('tspan')
+      .attr('x', x)
+      .attr('y', y)
+      .attr('id', 'BaseSpan')
+      .attr('dy', dy + 'em')
+      .attr('data-', totalWord);
+
+    if (truncateLabel && totalWordLength > noOfCharsToTruncate) {
+      text
+        .append('tspan')
+        .attr('id', 'showDots')
+        .attr('x', x)
+        .attr('y', y)
+        .attr('dy', dy)
+        .attr('dx', padding + dx + 'em')
+        .text(truncatedWord);
+    } else {
+      text
+        .attr('text-align', 'start')
+        .append('tspan')
+        .attr('id', 'LessLength')
+        .attr('x', isRtl ? xValue : x)
+        .attr('y', y)
+        .attr('dx', padding + dx + 'em')
+        .text(totalWord);
+    }
+  });
+}
+/**
+ * This function is calculating the length of longest Y axis label in px ,so that we are able to
+ * create the cartesian chart by shifting that many points to the right/left.
+ * @param points
+ * @returns
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function calculateLongestYAxisLabel(points: IHorizontalBarChartWithAxisDataPoint[], theme: ITheme): number {
+  let maxLabelLength = 0;
+  points.forEach((point: IHorizontalBarChartWithAxisDataPoint) => {
+    const canvas = document.createElement('canvas');
+    const ctx = canvas.getContext('2d');
+    ctx!.font = theme.fonts.tiny.key;
```

Comment:
is this way accurate. We should try using getComputedTextLength

Reply:
I have improved the accuracy. Please have  a look.

Reply:
We are using getComputedTextLength in other places. Why did you choose a new way to compute label length.